pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -44,25 +45,109 @@ type isImmutableFunc func(runtime.Object) bool
 // objectCacheItem is a single item stored in objectCache.
 type objectCacheItem struct {
 	refCount  int
-	store     cache.Store
+	store     *cacheStore
+	reflector *cache.Reflector
+
 	hasSynced func() (bool, error)
 
-	// lock is protecting from closing stopCh multiple times.
-	lock   sync.Mutex
-	stopCh chan struct{}
+	waitGroup sync.WaitGroup
```

Comment:
add a comment like:

waitGroup is used to ensure that there won't be two concurrent calls to reflector.Run